packages/lodestar/src/chain/validation/syncCommittee.ts

```diff
@@ -0,0 +1,94 @@
+import {CachedBeaconState, computeSubnetsForSyncCommittee, fast} from "@chainsafe/lodestar-beacon-state-transition";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {altair} from "@chainsafe/lodestar-types";
+import {IBeaconDb} from "../../db";
+import {ISyncCommitteeJob, SyncCommitteeError, SyncCommitteeErrorCode} from "../errors/syncCommitteeError";
+import {IBeaconChain} from "../interface";
+
+export async function validateGossipSyncCommittee(
+  config: IBeaconConfig,
+  chain: IBeaconChain,
+  db: IBeaconDb,
+  syncCommitteeJob: ISyncCommitteeJob,
+  subnet: number
+): Promise<void> {
+  const {syncCommittee, validSignature} = syncCommitteeJob;
+  // The signature's slot is for the current slot
+  if (chain.clock.currentSlot < syncCommittee.slot) {
+    throw new SyncCommitteeError({
+      code: SyncCommitteeErrorCode.FUTURE_SLOT,
+      currentSlot: chain.clock.currentSlot,
+      syncCommitteeSlot: syncCommittee.slot,
+      job: syncCommitteeJob,
+    });
+  }
+
+  if (chain.clock.currentSlot > syncCommittee.slot) {
+    throw new SyncCommitteeError({
+      code: SyncCommitteeErrorCode.PAST_SLOT,
+      currentSlot: chain.clock.currentSlot,
+      syncCommitteeSlot: syncCommittee.slot,
+      job: syncCommitteeJob,
+    });
+  }
+
+  // The block being signed over has been seen
+  if (!chain.forkChoice.hasBlock(syncCommittee.beaconBlockRoot)) {
+    throw new SyncCommitteeError({
+      code: SyncCommitteeErrorCode.UNKNOWN_BEACON_BLOCK_ROOT,
+      beaconBlockRoot: syncCommittee.beaconBlockRoot as Uint8Array,
+      job: syncCommitteeJob,
+    });
+  }
+
+  // There has been no other valid sync committee signature with same slot and validatorIndex
+  if (db.seenSyncCommiteeCache.hasSyncCommitteeSignature(syncCommittee)) {
+    throw new SyncCommitteeError({
+      code: SyncCommitteeErrorCode.SYNC_COMMITTEE_ALREADY_KNOWN,
+      root: config.types.altair.SyncCommitteeSignature.hashTreeRoot(syncCommittee),
+      job: syncCommitteeJob,
+    });
+  }
+
+  const headState = chain.getHeadState();
+  // TODO: https://github.com/ChainSafe/ssz/issues/102
+  // validate validator being part of the current sync committee
+  // const validatorPubkey = headState.validators[syncCommittee.validatorIndex].pubkey;
+  // const currentSyncCommittee = (headState as CachedBeaconState<altair.BeaconState>).currentSyncCommittee;
+  // const syncCommitteePubkeys = Array.from(readonlyValues(currentSyncCommittee.pubkeys));
+  // if (
+  //   syncCommitteePubkeys.findIndex((pubkey) => config.types.phase0.BLSPubkey.equals(pubkey, validatorPubkey)) === -1
+  // ) {
+  //   throw new SyncCommitteeError({
+  //     code: SyncCommitteeErrorCode.VALIDATOR_NOT_IN_SYNC_COMMITTEE,
+  //     validatorIndex: syncCommittee.validatorIndex,
+  //     job: syncCommitteeJob,
+  //   });
+  // }
+
+  // validate subnet
+  const expectedSubnets = computeSubnetsForSyncCommittee(
```

Comment:
Similar to above, what the cost of this utility `computeSubnetsForSyncCommittee`? Ensure it's efficient by maybe adding some performance tests to ensure we don't have the same issues as with prepareBeaconCommitteeSubnet

Reply:
I'll add: `// TODO: cache syncCommitteeIndices in CachedBeaconState`